substrate/frame/revive/src/evm/runtime.rs

```diff
@@ -0,0 +1,636 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//! Runtime types for integrating `pallet-revive` with the EVM.
+use crate::{
+	evm::api::{TransactionLegacySigned, TransactionLegacyUnsigned},
+	AccountIdOf, AddressMapper, BalanceOf, MomentOf, Weight, LOG_TARGET,
+};
+use codec::{Decode, Encode};
+use frame_support::{
+	dispatch::{DispatchInfo, GetDispatchInfo},
+	traits::ExtrinsicCall,
+};
+use pallet_transaction_payment::OnChargeTransaction;
+use scale_info::TypeInfo;
+use sp_arithmetic::Percent;
+use sp_core::{Get, U256};
+use sp_runtime::{
+	generic::{self, CheckedExtrinsic},
+	traits::{
+		self, Checkable, Dispatchable, Extrinsic, ExtrinsicMetadata, IdentifyAccount, Member,
+		SignedExtension,
+	},
+	transaction_validity::{InvalidTransaction, TransactionValidityError},
+	RuntimeDebug, Saturating,
+};
+
+use alloc::vec::Vec;
+
+/// The EVM gas price.
+/// We use a fixed value for the gas price.
+/// This let us calculate the gas estimate for a transaction with the formula:
+/// `estimate_gas = substrate_fee / gas_price`.
+pub const GAS_PRICE: u128 = 1_000u128;
```

Comment:
Need some more explanation what "we use" means here. I assume that means this is what we return from some RPC and this is also why this is public (it is used in the proxy).

Also why don't we use `1`?

Reply:
yes this is shared with the proxy, I moved it from the proxy to here so we can share the constant.
I was using 1 but then figured that this gas_price that we return from the rpc might be used by the wallet to adjust how much tip it wants to give to the tx. Wallet live in a world where they can use fractional number so 1 might still be fine.

Reply:
Yes but can you put that in the doc that this constant is used by the proxy to advertise it via the gas price rpc?